lib/rules/no-template-curly-in-string.js

```diff
@@ -22,19 +22,34 @@ module.exports = {
         schema: [],
 
         messages: {
-            unexpectedTemplateExpression: "Unexpected template string expression."
-        }
+            unexpectedTemplateExpression: "Unexpected template string expression.",
+            convertToTemplate: "Convert to ES6 `template`."
+        },
+        hasSuggestions: true
     },
 
     create(context) {
         const regex = /\$\{[^}]+\}/u;
+        const sourceCode = context.getSourceCode();
 
         return {
             Literal(node) {
                 if (typeof node.value === "string" && regex.test(node.value)) {
                     context.report({
                         node,
-                        messageId: "unexpectedTemplateExpression"
+                        messageId: "unexpectedTemplateExpression",
+                        suggest: [
+                            {
+                                messageId: "convertToTemplate",
+                                fix: fixer => {
+                                    const text = sourceCode.getText(node);
+                                    const textNoQuotes = text.slice(1, -1);
+                                    const newText = `\`${textNoQuotes}\``;
+
+                                    return fixer.replaceText(node, newText);
+                                }
```

Comment:
That's the main reason behind making it a "Suggestion" as opposed to a fix. I'm not completly sure of how they work via the command line, I assumed they each had their own behaviour, but if this is not the case, I can easily see why this would be a bad idea; automatic fixes shouldn't break stuff

Reply:
To avoid re-parsing the whole file, would it be valif to check the syntax by re-parsing the string by itself as a bare `ExpressionStatement`? How context-dependent is the template curly expression grammar?

Reply:
In [TemplateLiterals](https://tc39.es/ecma262/#prod-Expression), template curly expression can be any [Expression](https://tc39.es/ecma262/#prod-Expression), so parsing template literal's code as a program and checking if the parsing has successfully produced `Program` with a body consisting of a single `ExpressionStatement` would mostly work well.

There are some cases where this check could mistakenly allow a fix that does produce syntax errors due to the surrounding context, but those are probably all edge cases. For example:

```js
class C {
    x = "${arguments}";
}
```

There are also some edge cases where this check could mistakenly prevent a valid fix, but that's certainly acceptable.

```js
class C {
    foo() {
        "${super.foo()}";
    }
}
```